tests/yast2_cmd/yast_users.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# All of cases is based on the reference:
+# https://www.suse.com/documentation/sles-15/singlehtml/book_sle_admin/book_sle_admin.html#id-1.3.3.6.13.6.34
+#
+# Summary: manages user accounts
+#     Requirement: external NIS server "wotan.suse.de"
+#     Key Steps:
+#       - adds a new user with a password and verifies
+#       - changes the passwd of the new user and homedir, and deletes this user
+#       - binds a nis server and starts ypbind service, and then lists all users from NIS server
+# Maintainer: Jun Wang <jgwang@suse.com>
+#
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    select_console 'root-console';
+    zypper_call("in yast2-users yast2-nis-client ypbind", exitcode => [0, 102, 103, 106]);
+
+    # adds a new user with a password and homedir and verifies
+    # bsc#1143516: exit code is always NOT zero when the yast command works successfully, so once ZERO, it means a bug
+    validate_script_output("yast users add username=test_yast password=suse || echo BUG#1143516", sub { (m/^BUG#1143516$/m) ? (!record_soft_failure("bsc#1143516 for SLE15+: exit code is NOT zero")) : return 1 });
```

Comment:
Just use script_run (like [this](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/virtualization/yast_virtualization.pm#L25)) instead of this complicated one liner

Reply:
> Just use script_run (like [this](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/virtualization/yast_virtualization.pm#L25)) instead of this complicated one liner

Yes, I have changed it

Reply:
Thank you, could you use the same way also for following line ?

Reply:
> Thank you, could you use the same way also for following line ?

yes, the line needs to mark a soft failure for SLE12SP1 and SP2, and this
just run "yast users list local"  once.

Reply:
Only thing that worries me is your use of validate_script_output and use of echo when failed for match to prove the bug, it's just unnecessary complicated you can do the same with script_run, it can be also one liner or not, it's not important.

Reply:
yes, this line check the exit code of 'yast users list' to prove the bug, but it also check the output of 'yast users list' to check if the above 'yast users add' works well. If using 'script_run', I think I have to run 'yast users list' twice, feel this not good. Actually I thought about how to check the exit code and the output at the same time for a long time, and there seems to be no better way.

Reply:
ok, makes sense :+1:

Reply:
Thank you